test/PostgreSQL/Test.hs

```diff
@@ -1227,6 +1231,192 @@ testCommonTableExpressions = do
             pure res
         asserting $ vals `shouldBe` fmap Value [2..11]
 
+testPostgresqlLocking :: SpecDb
+testPostgresqlLocking =
+    -- These tests use a create a fork in order to run two transactions concurrently.
+    -- This is necessary in order to test the locking behaviour
```

Comment:
I could possibly abstract this forking behavior out into a function but it's a bit messy given the need for different queries and assertions. Please let me know if that would be strongly preferred though!